### PR TITLE
Documentation & scripts for the new ledger-core architecture

### PR DESCRIPTION
**Pull request by phaazon**

### CODE REVIEW

File: doc/account-synchronization.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        `"bitcoin"`. The list of identifiers should be available on the [crypto assets repository].
+      3. The list of addresses, comma-separated.
+      4. A list of parameters for the explorers to work with.
+    5. The result is then parsed as JSON by using a `LedgerApiParser` into a `TransactionsBulk`
+      depending on the type of coin.
+  2. The resulting batch is then inspected. For all transactions:
+    1. We look for _pending transactions_. If one is found, it is injected the transaction;
+      otherwise it’s erased.
+    2. The transaction is injected into the account and the last block is updated.
+	3. If the batch has no transaction, it means that the synchronization is over. We use that kind of
```
Comment (KhalilBellakrid) [inline doc/account-synchronization.md]: If you want to talk about the gap, you should just talk about HD wallet derivation, which is not related to a currency BTW and even omit the part that currencies have different "standards" (imposed by wallets/users and not protocol)

File: doc/account-synchronization.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    mechanism because of gap addresses: some currencies require a gap of a certain number of
+    deterministic addresses to test. When no transaction refers to them, no one will (until a new
+    transaction is inserted) and then we can safely end the synchronization.
+  4. If the batch had transactions, we have to recompute the next index batch and generate new
+    observable addresses. That can be pictured as a recursive algorithm, then.
+  5. The set of _to-drop_ transactions is traversed and transactions are dropped. Those are
```
Comment (KhalilBellakrid) [inline doc/account-synchronization.md]: Same here, if you want to talk about that, you should explain how we evaluate that we can drop a transaction (talk about mempool etc ...)

File: doc/architecture.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+The current state (_2019/08/06_) of the codebase is a bit complex. There is the existing, _in-prod_
+library that we call `lib-ledger-core`. That library has a convoluted architecture:
+
+  - The `$ROOT/core` directory contains all the files required to compile the C++ library and
+    generate the foreign interfaces.
+    - `$ROOT/core/idl` contains the IDL (Intermediatee Definition Language) files that defines the
+      public foreign interface. What it means is that all the symbols (data types, constants,
+      functions, etc.) defined in those files are the ones that will be publicly exported and
+      available when using the library. If you use the _React Native_ version of the library, you
```
Comment (KhalilBellakrid) [inline doc/architecture.md]: Which file are you talking about ?

File: doc/architecture.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+      will get a public Javascript file that will export the symbols you can find in the IDLs.
+    - `$ROOT/core/lib` vendors most of our dependencies. Currently, the libraries needed at runtime
+      are located in that directory. The rest of our dependencies (for instance, the CMake
+      toolchains or [djinni]) are located in specific locations. Consider reading the [.gitmodules]
+      files for further information.
+    - `$ROOT/core/src` contains all the C++ code required to implement the library and its bindings.
+      In that folder, there are two important sub-directories you must undertand the usecase:
```
Comment (KhalilBellakrid) [inline doc/architecture.md]: Some typo + reformulation ?

File: doc/architecture.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+  3. Configure your setup based on your platform. For instance, here, on _macOSX_:
+    - `cmake -DCMAKE_INSTALL_PREFIX=/path/to/qt5 -CMAKE_BUILD_TYPE=Debug ..`
+  4. Compile the library.
+    - `make -j8`
+  5. If you want to test locally with a product, like Ledger Live Desktop or Mobile, you need to
+    have cloned their repository beforehand and place them at the same level as `lib-ledger-core`.
+    You also need to have cloned the appropriate binding projects.
+      - For Ledger Live Desktop, you want [lib-ledger-core-node-bindings].
+      - For Ledger Live Mobile, you want [lib-ledger-core-react-native-bindings].
+      - For Vault, no need to generate anything but you have to recompile the library with JNI
```
Comment (KhalilBellakrid) [inline doc/architecture.md]: Of course you need to generate files, there is a `package.scala` generated

File: doc/architecture.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    `$ROOT/tools/generateBindingsRN.sh`, depending on your target.
+
+Refer to the appropriate projects for further information on how to build and use them.
+
+[Core team]: https://github.com/orgs/LedgerHQ/teams/core-library
+[Ledger]: https://www.ledger.com
+[djinni]: https://github.com/LedgerHQ/djinni
+[.gitmodules]: .gitmodules
+[lib-ledger-core-node-bindings]: https://github.com/LedgerHQ/lib-ledger-core-node-bindings
+[lib-ledger-core-react-native-bindings]: https://github.com/LedgerHQ/lib-ledger-core-react-native-bindings
```
Comment (KhalilBellakrid) [inline doc/architecture.md]: We should add the wallet daemon too in this case

File: doc/coin-migration.md [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+  - `ledger-core-<coin-name>` is a library that provides Core support for `<coin-name>`. For
+    instance, `ledger-core-ripple` provides the Ripple blockchain via our Core library.
+  - `ledger-core-factories` is a special library that lists all currently supported coins. It is
+    not necessarily mandatory in order to use the Core library but highly recommended to get started
+    with minimal amount of effort as it contains all objects to automatically synchronize a given
+    account without you to care about the dependent objects.
+
+## Migration foreword
+
+Before doing anything else, you need to know a script exists to automate **a lot** of things for
```
Comment (KhalilBellakrid) [inline doc/coin-migration.md]: Missing `that`

File: doc/account-synchronization.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+        and `v3` is an implementation detail (different internal architecture and infrastructure),
+        for most part.
+      2. The identifier of the blockchain network we target. For instance, for bitcoin, we use
+        `"bitcoin"`. The list of identifiers should be available on the [crypto assets repository].
+      3. The list of addresses, comma-separated.
+      4. A list of parameters for the explorers to work with.
+    5. The result is then parsed as JSON by using a `LedgerApiParser` into a `TransactionsBulk`
+      depending on the type of coin.
+  2. The resulting batch is then inspected. For all transactions:
+    1. We look for _pending transactions_. If one is found, it is injected the transaction;
```
[the remark this answers is not in the document]
Comment (phaazon) [reply doc/account-synchronization.md]: Pending transactions are regarded first when iterating on received batches to update them (if they’re still pending).

File: doc/architecture.md (hunk 3 above)
Comment (phaazon) [reply doc/architecture.md]: Generated NPM modules’ files.

File: doc/coin-migration.md (hunk 7 above)
Comment (phaazon) [reply doc/coin-migration.md]: Where?

**Comment by phaazon** [comment]
Closing to update and clean everything.

**Pull request closed without merging**